build/helper/metadata_add_all.py

```diff
@@ -627,13 +628,20 @@ def _add_enum_value_python_name(enum_info, config):
     # '_' only means the name starts with a number
     if len(prefix) > 0 and prefix != '_':
         for v in enum_info['values']:
+            if v['user_set_python_name']:
+                continue
             assert v['python_name'].startswith(prefix), '{} does not start with {}'.format(v['name'], prefix)
             v['prefix'] = prefix
             v['python_name'] = v['python_name'].replace(prefix, '')
 
     # Now we need to look for common suffixes
     # Using the slow method of reversing a string for readability
-    rev_names = [''.join(reversed(v['python_name'])) for v in enum_info['values']]
+    # We do not include hardcoded python names when looking for common suffixes
```

Comment:
Here you use `hardcoded` as a synonym to `user_set`. Reduce ambiguity by using one term to refer to one concept everywhere.

Reply:
FWIW I like the word "explicit" for this. It means (in my mind) that it's there written in metadata.
"User set" is a bit ambiguous because it's not super clear who the user is.

Reply:
I've stopped using the word hardcoded and made use of the word explicit, instead, as part of my rewrite.
I eliminated the `user_set_python_name` key. We'll do processing on the temporary key, `'_python_name'`, instead, so we won't need a special key to track whether we should expand the name.